engine/src/main/java/org/pentaho/di/core/gui/BasePainter.java

```diff
@@ -0,0 +1,511 @@
+    };
+
+    // Draw shadow around note?
+    if ( notePadMeta.isDrawShadow() ) {
+      int s = shadowSize;
+      int[] shadowa = new int[] {
+        note.x + s, note.y + s, // Top left
+        note.x + width + 2 * margin + s, note.y + s, // Top right
+        note.x + width + 2 * margin + s, note.y + height + s, // bottom right 1
+        note.x + width + s, note.y + height + 2 * margin + s, // bottom right 2
+        note.x + s, note.y + height + 2 * margin + s // bottom left
+      };
+      gc.setBackground( EColor.LIGHTGRAY );
+      gc.fillPolygon( shadowa );
+    }
+    gc.setBackground( notePadMeta.getBackGroundColorRed(), notePadMeta.getBackGroundColorGreen(), notePadMeta
+      .getBackGroundColorBlue() );
+    gc.setForeground( notePadMeta.getBorderColorRed(), notePadMeta.getBorderColorGreen(), notePadMeta
+      .getBorderColorBlue() );
+
+    gc.fillPolygon( noteshape );
+    gc.drawPolygon( noteshape );
+
+    if ( !Utils.isEmpty( notePadMeta.getNote() ) ) {
+      gc.setForeground( notePadMeta.getFontColorRed(), notePadMeta.getFontColorGreen(), notePadMeta
+        .getFontColorBlue() );
+      gc.drawText( notePadMeta.getNote(), note.x + margin, note.y + margin, true );
+    }
+
+    notePadMeta.width = width; // Save for the "mouse" later on...
+    notePadMeta.height = height;
+
+    if ( notePadMeta.isSelected() ) {
+      gc.setLineWidth( 1 );
+    } else {
+      gc.setLineWidth( 2 );
+    }
+
+    // Add to the list of areas...
+    //
+    if ( !shadow ) {
+      areaOwners.add( new AreaOwner( AreaType.NOTE, note.x, note.y, width, height, offset, subject, notePadMeta ) );
+    }
+  }
+
+  protected int translateTo1To1( int value ) {
+    return Math.round( value / magnification );
+  }
+
+  protected int translateToCurrentScale( int value ) {
+    return Math.round( value * magnification );
+  }
+
+  protected Point real2screen( int x, int y ) {
+    Point screen = new Point( x + offset.x, y + offset.y );
+
+    return screen;
+  }
+
+  protected Point getThumb( Point area, Point transMax ) {
+    Point resizedMax = magnifyPoint( transMax );
+
+    Point thumb = new Point( 0, 0 );
+    if ( resizedMax.x <= area.x ) {
+      thumb.x = 100;
+    } else {
+      thumb.x = (int) Math.floor( 100d * area.x / resizedMax.x );
+    }
+
+    if ( resizedMax.y <= area.y ) {
+      thumb.y = 100;
+    } else {
+      thumb.y = (int) Math.floor( 100d * area.y / resizedMax.y );
+    }
+
+    return thumb;
+  }
+
+  protected Point magnifyPoint( Point p ) {
+    return new Point( Math.round( p.x * magnification ), Math.round( p.y * magnification ) );
+  }
+
+  protected Point getOffset( Point thumb, Point area ) {
+    Point p = new Point( 0, 0 );
+
+    if ( hori == null || vert == null ) {
+      return p;
+    }
+
+    Point sel = new Point( hori.getSelection(), vert.getSelection() );
+
+    if ( thumb.x == 0 || thumb.y == 0 ) {
+      return p;
+    }
+
+    p.x = Math.round( -sel.x * area.x / thumb.x / magnification );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).